### PR TITLE
Component/list

### LINKED ISSUE #79: Create generic list component that can be used for different resources
```
import React from "react";

const List = ({ Component, data, title }) => {
  return (
    <>
      <h1>{title}</h1>
      <StyledList>
        {data.map((props) => (
          <Component {...{ ...props }} />
        ))}
      </StyledList>
    </>
  );
};

const StyledList = styled.ul`
  display: grid;
  gap: 20px;
`;

export default List;
```

https://www.figma.com/file/gGwSe7ilZCUmAIaJyUc4iY/Bristol-Tech-Fair-Design-System?node-id=308%3A88

### PR DESCRIPTION
**Pull request by catherinehoward**
Closes #79 

This pull request adds a generic list component.

### CODE REVIEW

**Comment by catherinehoward** [comment]
I've made a start on this. Can't get it to work at the moment. I'm getting the following error message:

ERROR in
src\components\List\List.jsx
  Line 11:11:  Missing "key" prop for element in iterator  react/jsx-key

File: src/components/List/List.jsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { ListWrapper, StyledList } from './List.styles';
+
+const List = ({ Component, data, title }) => {
+  return (
+    <ListWrapper>
+      <h1>{title}</h1>
+      <StyledList>
+        {data.map((props) => (
```
Comment (RobTF9) [inline src/components/List/List.jsx]:
```
   <Component { ...{ ...props, key: props._id }} />
```

File: src/components/List/List.styles.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import styled from 'styled-components';
+
+export const ListWrapper = styled.div`
+  padding: var(--small-space);
+`;
+
+export const StyledList = styled.ul`
+  display: grid;
+  gap: 20px;
```
Comment (RobTF9) [inline src/components/List/List.styles.js]: Use a space variable e.g. `--medium-space`

**Review by RobTF9 — CHANGES_REQUESTED** [review]

**Review by RobTF9 — APPROVED** [review]
Looks good to me 👍

File: src/components/List/List.stories.jsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      />
+    </li>
+    <li>
+      <ClubCard
+        coverUrl={data[1].coverUrl}
+        title={data[1].title}
+        to={data[1].to}
+      />
+    </li>
+  </Router>
```
Comment (RobTF9) [inline src/components/List/List.stories.jsx]: Hey, just re=checked and this example isn't right. You don't need the ClubCard components in the `li` tags, just pass the data array as a prop to the List.

**Review by RobTF9 — CHANGES_REQUESTED** [review]

File: src/components/List/List.stories.jsx (hunk 3 above)
Comment (catherinehoward) [reply src/components/List/List.stories.jsx]: Thanks for checking @RobTF9 . I've made the change you suggested and added some media queries: grid of 4 for desktop and 2 for tablets. All displaying correctly. Should the h1 be in the page components instead?

**Comment by admtomas** [comment]
List is working fine. We need resolve `<h>`  tag do we really need it or we can get away with it.

**Comment by catherinehoward** [comment]
I've removed the `<h1>` tag. Can someone please review the code. Thanks

**Review by admtomas — APPROVED** [review]
At the moment i do not see any issues.

**Comment by catherinehoward** [comment]
Merging is still blocked. @RobTF9 do you need to approve this as you requested the changes originally?

**Review by RobTF9 — APPROVED** [review]

**Pull request merged**